### ISSUE TITLE
Crash - SDWebImageDownloaderOperation.m line 219

### ISSUE THREAD

**Issue by guidev** [SDWebImage/SDWebImage#3106]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios 
 Platform Version        | 14.x
 SDWebImage Version      | 5.9.3
 Integration Method      | cocoapods
 Xcode Version           | Xcode 12.0.1
 Repro rate              | sometimes

### Issue Description and Steps

Hello,
I'm seeing a lot of crashes in Firebase due to SDWebImageDownloaderOperation.m line 219. It's roughly 2% of users.

Here are the screenshots from Firebase:

<img width="1356" alt="Screenshot 2020-10-14 at 12 04 33" src="https://user-images.githubusercontent.com/1105813/95975047-fc9a3a00-0e15-11eb-8555-ec07bd9ae2ed.png">
<img width="1300" alt="Screenshot 2020-10-14 at 12 04 58" src="https://user-images.githubusercontent.com/1105813/95975056-fe63fd80-0e15-11eb-8da7-6648c5ee72b4.png">

**Comment by Panajev** [comment]
Are this new to 5.9.3, did you use 5.9.2 before without issues?

**Comment by guidev** [comment]
> Are this new to 5.9.3, did you use 5.9.2 before without issues?

No, I have the same issue on 5.92

**Comment by dreampiggy** [comment]
Need more context about this, such as iOS platform version, or usage code which trigger this issue.

**Comment by aweissbr** [comment]
We have the same crashes in our Apps since Releases last week. The most of them are:

```swift

relating to SDWebImage
----------
0 libobjc.A.dylib		objc_msgSend + 16
6 CFNetwork		        -[__NSCFURLSessionTask resume] + 104
7 SDWebImage 		SDWebImageDownloaderOperation.m - Zeile 211
					-[SDWebImageDownloaderOperation start] + 211
other...
----------
0 libobjc.A.dylib 	objc_retain + 16
1 Foundation 		NSKeyValueWillChange + 404
2 Foundation 		-[NSObject(NSKeyValueObservingPrivate) _changeValueForKeys:count:maybeOldValuesDict:maybeNewValuesDict:usingBlock:] + 432

----------

0 CoreFoundation			CFDictionaryGetValue + 88
1 CoreFoundation    		_NSSetLongLongValueAndNotify + 92
9 libsystem_pthread.dylib 	start_wqthread + 8

```
from Crashlytics

Please check this link:  https://developer.apple.com/forums/thread/664233

**Comment by dreampiggy** [comment]
Looks like KVO crash. Does the usage of SDWebImage itself has something wrong ?

**Comment by dreampiggy** [comment]
Seems can be solved by #3146
